haddock-api/src/Haddock/Types.hs

```diff
@@ -640,9 +651,49 @@ data SinceQual
 -- A monad which collects error messages, locally defined to avoid a dep on mtl
 
 
-type ErrMsg = String
-type ErrMsgM = Writer [ErrMsg]
+type ErrMsg = Builder
 
+errMsgFromString :: String -> ErrMsg
+errMsgFromString = fromString
+
+errMsgToString :: ErrMsg -> String
+errMsgToString = Text.unpack . Text.decodeUtf8 . BSL.toStrict . toLazyByteString
+
+errMsgUnlines :: [ErrMsg] -> ErrMsg
+errMsgUnlines = List.foldl' (\acc x -> acc <> (x <> charUtf8 '\n')) mempty
```

Comment:
Does this also do a reverse? If that’s not intended I’d suggest using an intersperse.

Reply:
Good call, thanks for the suggestion!